### PR TITLE
CSM-2855: Displaying an error if empty folder is added

### PR DESCRIPTION
**Pull request by satwikgoyal**
### 🔗 Linked issue

- https://energysage.atlassian.net/browse/CSM-2855

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Earlier we wouldn't trigger/display any message when the user tried to upload  a folder but now we do

### 🥼 Testing

- Run `es-ds` in local using `make dev`. Adding a folder would trigger/emit `fileTypeError`

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

File: es-vue-base/src/lib-components/EsFileInput.vue [hunk 1]
```diff
@@ -201,6 +201,10 @@ export default {
             this.pickedItems = [];
         },
         async verifyMimeType(file) {
+            // If a folder then trigger fileTypeError
+            if (file.type === '') {
+                this.$emit('fileTypeError', file.name);
```
Comment (zacharyjarnagin) [inline es-vue-base/src/lib-components/EsFileInput.vue:206]: You could make this something like `fileIsAFolderError` or maybe emit a boolean flag to indicate if the error is due to the file being read as a folder?

Comment (satwikgoyal) [reply es-vue-base/src/lib-components/EsFileInput.vue]: thank you for your comments, made the changes

File: es-vue-base/src/lib-components/EsFileInput.vue [hunk 2]
```diff
@@ -206,6 +206,10 @@ export default {
             this.pickedItems = [];
         },
         async verifyMimeType(file) {
+            // If a folder then trigger fileTypeError
+            if (file.type === '') {
+                this.$emit('fileIsAFolderError');
```
Comment (zacharyjarnagin) [inline es-vue-base/src/lib-components/EsFileInput.vue:211]: Can you add a listener in the ES-DS page to showcase this? https://github.com/EnergySage/es-ds/blob/main/es-design-system/pages/molecules/es-file-input.vue

File: es-vue-base/src/lib-components/EsFileInput.vue [hunk 3]
```diff
@@ -206,6 +206,10 @@ export default {
             this.pickedItems = [];
         },
         async verifyMimeType(file) {
+            // If a folder then trigger fileTypeError
```
Comment (gigiwargin) [inline es-vue-base/src/lib-components/EsFileInput.vue:209]: Do you mean "If not a folder"?

**Review by gigiwargin — APPROVED** [review]
This looks great to me! Tested it out and works as expected! Just a question about the comment but definitely not blocking 🥳 🥳

File: es-vue-base/src/lib-components/EsFileInput.vue (hunk 3 above)
Comment (satwikgoyal) [reply es-vue-base/src/lib-components/EsFileInput.vue:209]: Ahh it should be "if an empty folder" resolving it. Thank you for noticing this!

**Review by gigiwargin — APPROVED** [review]
LGTM! Ty Satwik 🥳 🚀

**Comment by tomleo** [comment]
@zacharyjarnagin ok to merge this?

**Comment by zacharyjarnagin** [comment]
It sure is!

**Pull request merged**